### ISSUE TITLE
About FLERT experiments

### ISSUE THREAD

**Issue by matirojasg** [flairNLP/flair#2732]
Hi! I have been using the approach proposed in FLERT, and at least in the fine-tuning experiments, it has given me excellent results, thank you.

My question is the following. I am performing an empirical study comparing contextualized embeddings obtained from Flair, BERT, and FLERT. However, when I want to incorporate contextualized embeddings from FLERT in the LSTM-CRF architecture the results are quite low. Even in some of the categories, the result of the micro f1-score is equal to 0, which does not occur with the Flair model, where results are greater than 0.9.

The code I am using is the following:

```
embeddings = TransformerWordEmbeddings(model="PlanTL-GOB-ES/roberta-base-biomedical-es",
                                       layers="-1",
                                       subtoken_pooling="first",
                                       fine_tune=True,
                                       use_context=True,
                                       )

# 5. initialize sequence tagger

tagger = SequenceTagger(hidden_size=128,
                        rnn_layers=3,
                        embeddings=embeddings,
                        tag_dictionary=label_dict,
                        tag_type='ner',
                        use_crf=True,
                        use_rnn=True,
                        reproject_embeddings=False,
                        )


# 6. initialize trainer
trainer = ModelTrainer(tagger, corpus)

# 7. start training
trainer.fine_tune('',
                  learning_rate=5.0e-6,
                  mini_batch_size=4,
                  mini_batch_chunk_size=1  # remove this parameter to speed up computation if you have a big GPU
                  )
```

Is this the way to incorporate contextualized FLERT embeddings to a BiLSTM-CRF or not? Are those hyperparameters ok, or which ones would you recommend me to use?

**Comment by matirojasg** [comment]
I am performing the experiments on a biomedical dataset in Spanish.

**Comment by alanakbik** [comment]
Hello @matirojasg the likely problem is that fine-tuning and the RNN don't go well together. We found that if you want to use an RNN, you should freeze the Transformer embeddings (`fine_tune=False`), use all layers (`layers="all"`), and use the normal `train()` routine with default parameters (default learning rate, optimizer, batch size, etc.) instead of `fine_tune()`. 

See our paper for more info: https://arxiv.org/abs/2011.06993 

So try:

```python
embeddings = TransformerWordEmbeddings(model="PlanTL-GOB-ES/roberta-base-biomedical-es",
                                       layers="all",
                                       subtoken_pooling="first",
                                       fine_tune=False,
                                       use_context=True,
                                       )

# 5. initialize sequence tagger

tagger = SequenceTagger(hidden_size=256,
                        rnn_layers=1,
                        embeddings=embeddings,
                        tag_dictionary=label_dict,
                        tag_type='ner',
                        use_crf=True,
                        use_rnn=True,
                        reproject_embeddings=True,
                        )


# 6. initialize trainer
trainer = ModelTrainer(tagger, corpus)

# 7. start training
trainer.train('',
                  )
```

**Comment by alanakbik** [comment]
Closing, but feel free to reopen if you have more questions!
